### PR TITLE
Refactor micro-ROS pages

### PR DESCRIPTION
**Pull request by pablogs9**
This PR add a micro-ROS page under Vulcanexus Developer tools and refactor the order of the tutorials

https://eprosima.easyredmine.com/issues/16188

### CODE REVIEW

**Review by Acuadros95 — APPROVED** [review]
LGTM

**Comment by pablogs9** [comment]
I have refactored the sections "Getting started", "Build System Components" and "User API" into a general "micro-ROS Documentation" at the Overview level.

PTAL @Acuadros95 @rsanchez15 @EduPonz

**Review by Acuadros95 — APPROVED** [review]

**Review by rsanchez15 — APPROVED** [review]
LGTM

File: docs/rst/introduction/developer_tools.rst [hunk 1]
```diff
@@ -20,7 +20,7 @@ Developer Tools
 - :ref:`ROS 2 Monitor <vulcanexus_monitor>`: graphical desktop application to monitor *ROS 2* communications.
 - :ref:`Fast DDS Statistics Backend <fastdds_statistics_backend>`: *ROS 2 Monitor*'s backend. Can be leveraged with other monitoring frontend applications (e.g. :ref:`Prometheus <tutorials_tools_prometheus>`).
 - :ref:`ROS 2 Shapes Demo <vulcanexus_shapes_demo>`: first demo application to understand the most used *ROS 2* Quality of Service (QoS) and test DDS and *ROS 2* communication.
-- :ref:`micro-ROS <tutorials_micro_introduction_microros>`: default embedded toolkit for *ROS 2* and *Vulcanexus*.
+- :ref:`micro-ROS <vulcanexus_microros>`: default embedded toolkit for *ROS 2* and *Vulcanexus*.
```
Comment (EduPonz) [inline docs/rst/introduction/developer_tools.rst:23]: I wonder whether microros is a developer tool

File: docs/rst/microros_documentation/index.rst [hunk 2]
```diff
@@ -1,18 +1,11 @@
-.. _tutorials_micro_introduction:
+.. _vulcanexus_microros:
```
Comment (EduPonz) [inline docs/rst/microros_documentation/index.rst:1]: Metapackages should reference this

**Review by EduPonz — CHANGES_REQUESTED** [review]

File: docs/rst/introduction/developer_tools.rst (hunk 1 above)
Comment (pablogs9) [reply docs/rst/introduction/developer_tools.rst]: Removed here: 8574e5be1f171fec4dc0e2d2026afb0abe1a32dc

File: docs/rst/microros_documentation/index.rst (hunk 2 above)
Comment (pablogs9) [reply docs/rst/microros_documentation/index.rst:1]: References here: 10d160e7c4386e998a7fe2916f09577ca18a87e4

File: docs/rst/introduction/introduction.rst [hunk 3]
```diff
@@ -46,7 +46,7 @@ Developer Tools
 - :ref:`ROS 2 Monitor <vulcanexus_monitor>`: graphical desktop application to monitor *ROS 2* communications.
 - :ref:`Fast DDS Statistics Backend <fastdds_statistics_backend>`: *ROS 2 Monitor*'s backend. Can be leveraged with other monitoring frontend applications (e.g. :ref:`Prometheus <tutorials_tools_prometheus>`).
 - :ref:`ROS 2 Shapes Demo <vulcanexus_shapes_demo>`: first demo application to understand the most used *ROS 2* Quality of Service (QoS) and test DDS and *ROS 2* communication.
-- :ref:`micro-ROS <tutorials_micro_introduction_microros>`: default embedded toolkit for *ROS 2* and *Vulcanexus*.
+- :ref:`micro-ROS <vulcanexus_microros>`: default embedded toolkit for *ROS 2* and *Vulcanexus*.
```
[the remark this answers is not in the document]
Comment (rsanchez15) [reply docs/rst/introduction/introduction.rst:49]: Also we should extend the sententence `Vulcanexus concepts including ROS 2 Documentation.` including something about micro-ROS in the index page.

[the remark this answers is not in the document]
Comment (pablogs9) [reply docs/rst/introduction/introduction.rst]: @EduPonz addressed
@rsanchez15 I cannot found where is this sentence in the repo

[the remark this answers is not in the document]
Comment (rsanchez15) [reply docs/rst/introduction/introduction.rst]: File `docs/index.rst`. 
In `Getting Started` section. 
Under `Overview` item.

**Review by EduPonz — APPROVED** [review]

File: docs/rst/introduction/introduction.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (pablogs9) [reply docs/rst/introduction/introduction.rst]: 9b01b864dca3ca28e84b72baaebf5c90090178ba

**Review by EduPonz — APPROVED** [review]

**Review by rsanchez15 — APPROVED** [review]

**Pull request merged**